routes/web.php

```diff
@@ -18,13 +18,18 @@
 Auth::routes();
 
 Route::get('/home', 'HomeController@index');
-Route::get('/projects/job-order/{id}/materials', 'Projects\\JobOrderController@materials');
+
+Route::get('projects/job-order/{id}/materials/create', 'Projects\\MaterialsController@create');
+Route::get('projects/job-order/{id}/materials/{matid}', 'Projects\\MaterialsController@update');
+Route::get('projects/job-order/{id}/materials/{matid}', 'Projects\\MaterialsController@destroy');
+Route::get('projects/job-order/{id}/materials/{matid}/show', 'Projects\\MaterialsController@show');
+Route::get('projects/job-order/{id}/materials/{matid}/edit', 'Projects\\MaterialsController@edit');
```

Comment:
try changing `$id` to `$job_id` and retain `$id` as default argument to the MaterialsController methods. And add `$job_id` as another argument.